src/power_grid_model/_utils.py

```diff
@@ -539,9 +539,28 @@ def is_columnar(component_data: ComponentData) -> bool:
     return not isinstance(component_data, np.ndarray)
 
 
-def is_nan_or_equivalent(array):
+def is_nan_or_default(x: np.ndarray) -> np.ndarray:
+    """
+    Check if elements in the array are NaN or equal to the min of its dtype.
+
+    Args:
+        x: A NumPy array to check.
+
+    Returns:
+        A boolean NumPy array where each element is True if the corresponding element in x is NaN
+        or min of its dtype, and False otherwise.
+    """
+    if x.dtype == np.float64:
+        return np.isnan(x)
+    if x.dtype in (np.int32, np.int8):
+        return x == np.iinfo(x.dtype).min
```

Comment:
you can also try to compare with the `nan`types defined in `power_grid_meta_data`. i understand if that's not going to work here, though. the good thing is that this is in `_utils` so we can do anything we want with the implementation

Reply:
That would be over complexing the issue.